data_model/src/wasm.rs

```diff
@@ -13,8 +13,10 @@ pub mod export {
         pub const WASM_ALLOC: &str = "_iroha_wasm_alloc";
         /// Exported function to deallocate memory
         pub const WASM_DEALLOC: &str = "_iroha_wasm_dealloc";
-        /// Name of the exported entry for smart contract or trigger execution
-        pub const WASM_MAIN: &str = "_iroha_wasm_main";
+        /// Name of the exported entry for smart contract execution
+        pub const SMART_CONTRACT_MAIN: &str = "_iroha_smart_contract_main";
+        /// Name of the exported entry for trigger execution
+        pub const TRIGGER_MAIN: &str = "_iroha_trigger_main";
```

Comment:
Maybe here we need to make a distinction along a different axis. This is related to the above discussion of language.  

We have three things that fit the definition of [smart contracts](https://en.wikipedia.org/wiki/Smart_contract): the single shot executable, or an on-chain program that is sent instead of a transaction, a hook which maps triggers to callbacks and an executor. 

So we could call the three as follows: transaction main, hook main and executor main, propagating the names throughout this pull request.